src/middleware/gzip_middleware.py

```
@@ -0,0 +1,327 @@
+
+        logger.debug(f"Processing gzip-compressed request to {path}")
+
+        body_parts = []
+        try:
+            while True:
+                message = await receive()
+                if message["type"] == "http.request":
+                    body_parts.append(message.get("body", b""))
+                    if not message.get("more_body", False):
+                        break
+        except (OSError, EOFError, RuntimeError) as e:
```

Flake8 (7.3.0)

[error] local variable 'e' is assigned to but never used

(F841)